packages/core/src/store/anchor-request-store.ts

```diff
@@ -132,7 +166,17 @@ export class AnchorRequestStore extends ObjectStore<StreamID, AnchorRequestData>
             `Anchor polling loop processed ${numEntries} entries from the AnchorRequestStore. Restarting loop.`
           )
           Metrics.observe(ANCHOR_POLLING_PROCESSED, numEntries)
-          await new Promise((resolve) => setTimeout(resolve, restartDelay))
+
+          // If we iterated over all entries in the store in less time than the minLoopDuration,
+          // sleep to avoid spamming the CAS.
+          const loopEndTime = new Date()
+          // jest sometimes does weird things with time so abs is necessary
+          const loopDurationMs = Math.abs(loopEndTime.getTime() - loopStartTime.getTime())
+          if (loopDurationMs < this.#minLoopDurationMs) {
```

Comment:
Can we change this to

```
const remainingLoopDuration = this.#minLoopDurationMs - loopDurationMs
if(remainingLoopDuration > this.#minLoopJitterMs) {
   await sleepOrAbort(remainingLoopDuration, this.#abortController.signal)
}```

That way if our remaining loop duration is something like 10ms, we don't bother awaiting.